internal/rust/parser/parser.go

```diff
@@ -0,0 +1,19 @@
+package parser
+
+//go:generate cargo build --release
```

Comment:
This won't work because of the permissions on the directory when this is included as a library. But, we may be able to do this. The key would be wrapping this command with either a script or a go binary that ensures the current directory is writable. If the current directory is not writable, it would attempt to make it writable so that we could write out the produced binary.

But I'm not sure if that will influence the hash that gets produced for `go.sum` since I am not sure how that is calculated.

Alternatively, we can choose a place in `/usr/local` or choose a location that is writable in the go cache and use that. I'll give that a try since it might be needed for the ARM builds if we choose to only include `x86_64` precompiled binaries in the module.